src/Controller/CRUDController.php

```diff
@@ -1360,8 +1360,27 @@ protected function handleXmlHttpRequestErrorResponse(Request $request, FormInter
         }
 
         $errors = [];
-        foreach ($form->getErrors(true) as $error) {
-            $errors[] = $error->getMessage();
+        $formName = $form->getName();
+
+        foreach ($form->getErrors(true) as $formError) {
+            $origin = $formError->getOrigin();
+
+            $fieldName = $origin->getName();
+            $name = '';
+
+            while ($origin = $origin->getParent()) {
+                if ($formName !== $origin->getName()) {
+                    $name = $origin->getName() . '_' . $name;
+                }
+            }
+
+            $fieldName = $formName . '_' . $name . $fieldName;
+
+            if (!in_array($fieldName, $errors)) {
+                $errors[$fieldName] = [];
+            }
+
+            $errors[$fieldName][] = $formError->getMessage();
```

Comment:
The error were `string[]` now it's `string[][]` so it's a BC break technically

Reply:
Yep, but one field can have multiple errors, do you have an idea on how to avoid this? IMO this method is unusable without those changes. Before this PR, you would get:

0 => 'Field cannot be empty'
1 => 'Field cannot be null',
2 => 'Field cannot be empty',

After:

'43543543_settings_url' => [
  'Field cannot be empty',
  'Field cannot be null',
],
'43543543_settings_name' => [
  'Field cannot be empty',
]